desktop/cmp/record/RecordActionBar.js

```diff
@@ -32,51 +33,59 @@ export class RecordActionBar extends Component {
 
     static propTypes = {
         /** RecordActions to clone or configs to create. */
-        actions: PT.arrayOf(PT.oneOfType([PT.object, PT.instanceOf(RecordAction)])).isRequired,
+        actions: PT.arrayOf(PT.oneOfType([PT.object, RecordAction])).isRequired,
         /** The data Record to associate with the actions. */
-        record: PT.instanceOf(Record),
+        record: PT.oneOfType([PT.object, Record]),
         /** Set to true to only show the action buttons when hovering over the action bar (or row when used in a grid). */
-        showOnHoverOnly: PT.bool
-    };
+        showOnHoverOnly: PT.bool,
 
-    actions = [];
+        selModel: PT.instanceOf(StoreSelectionModel),
 
-    constructor(props) {
-        super(props);
-        this.actions = props.actions
-            .filter(Boolean)
-            .map(it => new RecordAction(it));
-    }
+        /** Data to pass through to action callbacks */
+        actionMetadata: PT.object,
```

Comment:
Just thinking about it -- not sure whether it really makes sense for the public component for use in a toolbar the same object as the private object used by the column (i.e.. leaves us with the wonky "showOnHoverOnly" property)

Reply:
Didn't address this in my most recent changes, but I see the point re: the `showOnHoverOnly` prop, don't know that you would ever use that outside of the grid action col actionBar.

We could actually remove the prop entirely and have the actionCol partial conditionally add a css class to the element it returns in the render fn based on the `actionShowOnHoverOnly` column prop... thoughts?